src/plugins/intel_gna/transformations/decompose_concat.cpp

```diff
@@ -0,0 +1,104 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <openvino/cc/ngraph/itt.hpp>
+
+#include "transformations/decompose_concat.hpp"
+
+#include <ngraph/opsets/opset9.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/rt_info.hpp>
+#include <transformations/utils/utils.hpp>
+#include "backend/gna_limitations.hpp"
+
+
+using namespace ov::intel_gna::pass;
+using namespace ngraph;
+
+struct ConcatData {
+    size_t num_inputs;
```

Comment:
maybe it would be good to initialize `num_inputs` and `axis` with some default values?